### PR TITLE
Fix a race condition during initialization (#11444)

### LINKED ISSUE #11444: Handle Race When Initializing The P2P Service
# 🐞 Bug Report

### Description

Currently if we initialize a libp2p host without initializing the peer manager in time, this could 
cause a panic.

Bug Found By Antithesis.

### Has this worked before in a previous version?

N.A

## 🔬 Minimal Reproduction

It would be possible to trigger this for nodes running via a preknown/static identity at startup. If they are constantly dialed in from
inbound peers.

## 🔥 Error

```
panic: runtime error: invalid memory address or nil pointer dereference
segmentation violation code=0x1 addr=0x50 pc=0x14ed43e]

goroutine 222 [running]:
github.com/prysmaticlabs/prysm/v3/beacon-chain/p2p.(*Service).isPeerAtLimit(0xc000312900, 0x1)
/git/src/github.com/prysmaticlabs/prysm/beacon-chain/p2p/discovery.go:460 +0x3e
github.com/prysmaticlabs/prysm/v3/beacon-chain/p2p.(*Service).InterceptAccept(0xc000312900, {0x7f6d3a0b6d60, 0xc0037abb00})
/git/src/github.com/prysmaticlabs/prysm/beacon-chain/p2p/connection_gater.go:56 +0xa5
[github.com/libp2p/go-libp2p/p2p/net/upgrader.(*listener).handleIncoming(0xc0002101c0)
/root/go/pkg/mod/github.com/libp2p/go-libp2p@v0.20.3/p2p/net/upgrader/listener.go:87 +0x172
created by github.com/libp2p/go-libp2p/p2p/net/upgrader.(*upgrader).UpgradeListener
/root/go/pkg/mod/github.com/libp2p/go-libp2p@v0.20.3/p2p/net/upgrader/upgrader.go:111 +0x225
```

**What version of Prysm are you running? (Which release)**

`57b7e0b57252eacd79ea64b5a25c6ab32ee42da9`

### PR DESCRIPTION
**Pull request by dyng**
**What type of PR is this?**

Bug fix

**What does this PR do? Why is it needed?**

Prevent p2p service from panic when inbound connection received during initialization.

**Which issues(s) does this PR fix?**

Fixes #11444

### CODE REVIEW

**Review by prestonvanloon — DISMISSED** [review]
Nice find, thanks

**Comment by dyng** [comment]
I have fixed the failed test cases and a trivial typo.

File: beacon-chain/p2p/connection_gater_test.go [hunk 1]
```diff
@@ -96,6 +98,20 @@ func TestService_InterceptBannedIP(t *testing.T) {
 	}
 }
 
+func TestService_RejectInboundConnectionBeforeStarted(t *testing.T) {
```
Comment (nisdas) [inline beacon-chain/p2p/connection_gater_test.go:101]: Could you also add the positive case in where an inbound connection is allowed now that the service has started ?

**Review by nisdas — COMMENTED** [review]
Thanks for the fix ! Looks great, just had 1 request

File: beacon-chain/p2p/connection_gater_test.go (hunk 1 above)
Comment (dyng) [reply beacon-chain/p2p/connection_gater_test.go:101]: Good idea, I'll add it.

**Review by rauljordan — APPROVED** [review]

**Pull request merged**